### PR TITLE
Align parsing version to datamodel

### PR DESCRIPTION
**Pull request by samleeflang**
Older version was based on the parsed database records.
It means that new versions will work but older versions in the old datamodel might break.

### CODE REVIEW

File: src/main/java/eu/dissco/backend/service/AnnotationService.java [hunk 1]
```diff
@@ -119,12 +119,18 @@ public AnnotationResponse persistAnnotation(AnnotationRequest annotation, String
...
-    return new AnnotationResponse(response.get("id").asText(), response.get("version").asInt(),
-        annotation.get("type").asText(), annotation.get("motivation").asText(),
-        annotation.get("target"), annotation.get("body"), annotation.get("preferenceScore").asInt(),
+    return new AnnotationResponse(response.get("id").asText(),
+        response.get("version").asInt(),
+        annotation.get("type").asText(),
+        annotation.get("motivation").asText(),
+        annotation.get("target"),
+        annotation.get("body"),
+        annotation.get("preferenceScore").asInt(),
```
Comment (TomDijkema) [inline src/main/java/eu/dissco/backend/service/AnnotationService.java:128]: Much better to read

File: src/main/java/eu/dissco/backend/service/AnnotationService.java [hunk 2]
```diff
@@ -119,12 +119,18 @@ public AnnotationResponse persistAnnotation(AnnotationRequest annotation, String
...
         annotation.get("creator").asText(),
-        Instant.ofEpochSecond(annotation.get("created").asLong()), annotation.get("generator"),
-        Instant.ofEpochSecond(annotation.get("generated").asLong()), null);
+        Instant.ofEpochSecond(annotation.get("created").asLong()),
+        annotation.get("generator"),
+        Instant.ofEpochSecond(annotation.get("generated").asLong()),
```
Comment (TomDijkema) [inline src/main/java/eu/dissco/backend/service/AnnotationService.java:132]: Is the transfer from epoch to datetime already on the schedule or is it not relevant to this part?

File: src/main/java/eu/dissco/backend/service/AnnotationService.java [hunk 3]
```diff
@@ -134,19 +140,20 @@ public AnnotationResponse getAnnotationByVersion(String id, int version)
...
+        annotation.get("type").asText(),
+        annotation.get("motivation").asText(),
+        annotation.get("target"),
+        annotation.get("body"),
+        annotation.get("preferenceScore").asInt(),
+        annotation.get("creator").asText(),
+        Instant.ofEpochSecond(annotation.get("created").asInt()),
+        annotation.get("generator"),
+        Instant.ofEpochSecond(annotation.get("generated").asInt()),
+        result.get("deleted") == null ? null : Instant.ofEpochSecond(annotation.get("deleted").asInt())
```
Comment (TomDijkema) [inline src/main/java/eu/dissco/backend/service/AnnotationService.java:156]: I like this approach far more than just result

**Review by TomDijkema — APPROVED** [review]

File: src/main/java/eu/dissco/backend/service/AnnotationService.java (hunk 2 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/backend/service/AnnotationService.java:132]: Yes it is on the backlog.

File: src/main/java/eu/dissco/backend/service/AnnotationService.java (hunk 3 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/backend/service/AnnotationService.java:156]: Yes so the main thing is that the metadata is generic for each event object. So we have event (metadata) -> eventRecord (metadata of object (pid, version,created) -> Actual object. Based on the SubjectType (an event metadata field) you can know which object is in the eventRecord. So I think it is more generic and more machine readable. Although fieldnames will probably change when we align with Prov-O

**Pull request merged**